### DISCUSSION TITLE
Why gdscript cannot inherit Array or Dictionary?

### DISCUSSION THREAD

**Discussion by colorworlds** [godotengine/godot-proposals#5840] [Scripting]
The following code is prohibited:  
```
extends Array
extens Dictionary
```
This is very different from other languages.  
If you want to expand an array by yourself, you must let a new class hold the array, but you cannot use for to traverse the array, such as `for elem in AClassHasArray`.  
Why is it designed like this?  
This is very inconvenient at some times

**Comment by colorworlds** [comment]
A specific example is storing data as json.  

(This problem does not exist if resource scripts are used for storage, and json is selected only to facilitate network transmission)  

The data I store and read are `Dictionary`, but when I want data to have some other operations, I need to wrap the `Dictionary` as a class.  

There is no problem if there is only one player object.  

However, if the player also holds multiple equipment data, it will provide additional complexity.  

Because these equipment are similar to players, they are an object with their own functions and need to be packaged as a class.  

The more such nesting, the higher the complexity.   

Every object that is considered to exist independently will have two versions of data and class, and maintaining the consistency of the two versions of data is very painful.

**Comment by dalexeev** [comment]
`Array` is a `Variant` type, not an `Object`.

> but you cannot use for to traverse the array, such as `for elem in AClassHasArray`.

See https://docs.godotengine.org/en/stable/tutorials/scripting/gdscript/gdscript_advanced.html#custom-iterators.
